Various bugfixes

BikeMikeAU opened the pull request.

Commit:
Various bugfixes

Commit:
Update communication.py

BikeMikeAU commented on bots/communication.py:
```diff
@@ -153,7 +153,7 @@ def run(self):
                         # ~ max_nr_retry = int(self.channeldict['rsrv1']) if self.channeldict['rsrv1'] else 0
                         # ~ if max_nr_retry:
-                            # ~ domain = (self.channeldict['idchannel'] + u'_failure')[:35]
+                            # ~ domain = (self.channeldict['idchannel'] + '_failure')[:35]
```
im not sure about why this section is commented out but updated the code anyway in case it goes back in

BikeMikeAU replied:
The else block should go before finally, then un-comment it.
Logic here is to only report when max_nr_retry errors in a row. As soon as there is a successful connection the counter should be reset to zero

BikeMikeAU replied:
I also like the idea of max_nr_connect_tries; this will help with transient timeouts but not if a server is really down for some period.

BikeMikeAU replied:
fixed this, see commit https://github.com/eppye-bots/bots3.3/pull/2/commits/f2eec06213919b4b16d8002e1290b76de4f5554b

Commit:
Update communication.py

syntax error indent

Commit:
Update engine.py

fix ini key for daily logging to match botsinit.py
print to console all severe errors

eppye-bots commented on bots/engine.py:
```diff
@@ -196,6 +198,7 @@ def start():
 
         cleanup.cleanup(do_cleanup_parameter,userscript,scriptname)
     except Exception as msg:
+        print('Severe error:' + msg)
```
not needed, is via logger

BikeMikeAU replied:
Added these for quicker debugging while testing, it's good to see errors on the console

eppye-bots commented on bots/engine.py:
```diff
@@ -85,17 +85,19 @@ def start():
     try:
         botsinit.connect()
     except Exception as msg:
+        print('Error connecting to database:' + msg)
```
not needed, is via logger

BikeMikeAU replied:
Added these for quicker debugging while testing, it's good to see errors on the console

eppye-bots commented on bots/engine.py:
```diff
@@ -85,17 +85,19 @@ def start():
     try:
         botsinit.connect()
     except Exception as msg:
+        print('Error connecting to database:' + msg)
```
not needed, is via logger

BikeMikeAU replied:
Added these for quicker debugging while testing, it's good to see errors on the console

eppye-bots commented on bots/engine.py:
```diff
@@ -85,17 +85,19 @@ def start():
         sys.exit(1)
     else:
-        if botsglobal.ini.get('settings','log_file_number','') != 'daily':
+        if botsglobal.ini.get('settings','log_when',None) != 'daily':
```
not sure I understand this. there is no 'log_when'. if new, what does it do?
it would be good to have an option to log engine to one log file (per day), as you have suggested..

BikeMikeAU replied:
I thought you had added log_when? It is there in botsinit so I changed engine to use it also.
Probably changed because botsglobal.ini.getint('settings','log_file_number',10) gives an error because 'daily' is not an int.
Also means you can set daily AND number of days to keep.

BikeMikeAU replied:
I originally made this change years ago and probably posted to the mailing list, but only used log_file_number at that time. In my installations I keep daily logs and do daily reporting on those in botsengine postcleanup.

BikeMikeAU replied:
Needs to be added in bots.ini but comment out by default

#log_file_number: number of rotating log files. Value: number. Default: 10
log_file_number = 10
#for one merged log per day, set log_when = daily. Default: Each run uses it's own log file
#log_when = daily

BikeMikeAU replied:
could also use a setting in bots.ini to explicitly state the default; it will not affect operation of this setting.

#for one merged log per day, set log_when = daily. Default: Each run uses it's own log file
log_when = each run

eppye-bots commented on bots/engine.py:
```diff
@@ -85,17 +85,19 @@ def start():
         botsglobal.logger = botsinit.initenginelogging(process_name)
         atexit.register(logging.shutdown)
     except Exception as msg:
+        print('Error initialising logging:' + msg)
```
not needed, is via logger

BikeMikeAU replied:
but this error is during log init, so never gets logged!
(I had this error a long time ago and took me a while to track down what caused it... but can't remember what it was now. Maybe permissions on the log folder)

Commit:
reset failure counter after a successful connection

Commit:
Fix up maxconnectiontries logic in run()

1. match variable names to the settings - less confusing reading the code :-)
2. maxconnectiontries applies to both in and out channels.
3. fix the try-except-else-finally (i had to read python docs again... break in finally clause was no good)
4. new python 3 exception context adds confusion to the trace when there is a connection error; fix that to just re-raise the original (raise exc from exc)

BikeMikeAU reviewed (commented):
91d209a commit doesn't properly fix it, the next one does though
